### PR TITLE
Feature/kids 221/welcome screen

### PR DESCRIPTION
**Pull request by alexbejann**
Contains the Amplitude settings and event logging as in the amplitude dashboard.

### CODE REVIEW

**Review by Daniela510 — APPROVED** [review]
LGTM!

File: lib/what_good/pages/chat_page.dart [hunk 1]
```diff
@@ -115,6 +123,10 @@ class _ChatPageView extends StatelessWidget {
                     return OutlinedButton(
                       onPressed: () {
                         if (e.text == 'Build my avatar') {
+                          AnalyticsHelper.logEvent(
+                            eventName: AmplitudeEvent.buildAvatarClicked,
+                          );
+                          if (!context.mounted) return;
```
Comment (pavelpavelko) [inline lib/what_good/pages/chat_page.dart:129]: shall we track the event only if mounted (after the check)?

File: lib/what_good/providers/chat_provider.dart [hunk 2]
```diff
@@ -17,7 +19,10 @@ class ChatProvider with ChangeNotifier {
   String get name => _name;
 
   /// This method is called when the user presses the submitt button from keyboard
-  set name(String value) {
+  Future<void> setName(String value) async {
+    final prefs = await SharedPreferences.getInstance();
```
Comment (pavelpavelko) [inline lib/what_good/providers/chat_provider.dart:23]: Storing username in both providers (ProfilesProvider and ChatProvider) looks like functionality duplication. Shall we use single source of data in this case?

File: lib/what_good/utils/analytics_helper.dart [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  static Future<void> logEvent({
+    required AmplitudeEvent eventName,
+    Map<String, dynamic>? eventProperties,
+  }) async {
+    await Amplitude.getInstance().logEvent(
+      eventName.value,
+      eventProperties: eventProperties,
+    );
+
+    dev.log('${eventName.value} pressed');
```
Comment (pavelpavelko) [inline lib/what_good/utils/analytics_helper.dart:38]: shall we log (I mean dev.log) the eventProperties here as well?

File: lib/what_good/pages/chat_page.dart (hunk 1 above)
Comment (alexbejann) [reply lib/what_good/pages/chat_page.dart]: Actually, that's useless as the onPressed method is not marked as async anymore. Therefore, the async gaps risk is not existent.

File: lib/what_good/providers/chat_provider.dart (hunk 2 above)
Comment (alexbejann) [reply lib/what_good/providers/chat_provider.dart:23]: Isn't the welcome screen the first in the flow? If there are not any profiles?

File: lib/what_good/utils/analytics_helper.dart (hunk 3 above)
Comment (alexbejann) [reply lib/what_good/utils/analytics_helper.dart:38]: Sure, give me a moment.

File: lib/what_good/pages/chat_page.dart (hunk 1 above)
Comment (alexbejann) [reply lib/what_good/pages/chat_page.dart]: Check it again, i removed it after i left the comment

**Pull request merged**